### PR TITLE
Menus now work as their own instance.

### PR DESCRIPTION
**Pull request by Zach777**
Pause and HUD now work when instanced. They are not reliant on being globals.

### CODE REVIEW

File: Tree/Interface/MainMenu/MainMenu.gd [hunk 1]
```diff
@@ -3,46 +3,48 @@ extends CanvasLayer
...
 """
 
-onready var tabs: TabContainer = $"H/T"
+onready var tabs: TabContainer = $"Top/Tabs"
 
 
 func show() -> void:
+	#Show the menu. This is code from when this menu worked differently.
+	#In case we ever need to instance this menu from another scene we can use
+	#this to show it.
```
Comment (Reousa) [inline Tree/Interface/MainMenu/MainMenu.gd:12]: nit. Let's simplify comments to be just what's necessary to understand.
```
# Show the menu & it's children.
func show() -> void:
```

File: Tree/Interface/MainMenu/MainMenu.gd [hunk 2]
```diff
@@ -3,46 +3,48 @@ extends CanvasLayer
...
 			i.visible = true
-	
-	PauseMenu.set_openable(false)
-	Hud.set_active(false)
-
 
 func hide() -> void:
+	#Hide the menu. This is code from when this menu worked differently.
+	#In case we ever need to instance this menu from another scene we can use
+	#this to hide it.
```
Comment (Reousa) [inline Tree/Interface/MainMenu/MainMenu.gd:20]: nit. Let's simplify comments to be just what's necessary to understand.

File: Tree/Interface/Hud/Chat/Chat.gd [hunk 3]
```diff
@@ -55,20 +55,19 @@ func _unhandled_input(event: InputEvent) -> void:
...
 			_toggle_chat_window()
 
-
 func _on_LineEdit_text_entered(new_text: String) -> void:
 	if new_text != "":
-		var username : String = "[color=#DB7900]" + Options.username +  ":[/color]: "
 		
-		rpc("_append_text_to_chat", username + new_text )
+		#var username : String = "[color=#DB7900]" + Options.username +  ":[/color]: "
+		#rpc("_append_text_to_chat", username + new_text )
```
Comment (Reousa) [inline Tree/Interface/Hud/Chat/Chat.gd:62]: We'll keep these for now - we'll remove them once their replacement counterparts are ready.
Just letting you know

**Comment by Reousa** [comment]
@Zach777 Please do squash & rebase the commits after finishing.

File: Tree/Interface/MainMenu/MainMenu.gd (hunk 1 above)
Comment (Zach777) [reply Tree/Interface/MainMenu/MainMenu.gd]: I fixed the commit based on your suggestions.

**Pull request merged**